python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -417,43 +418,49 @@ def deserialize(cls, header, frames):
         return cls(obj, grouping, **kwargs)
 
     def _grouped(self):
-        grouped_keys, grouped_values, offsets = self._groupby.groups(self.obj)
-        grouped_values = self.obj.__class__._from_data(*grouped_values)
-        grouped_values._copy_type_metadata(self.obj)
+        grouped_key_cols, grouped_value_cols, offsets = self._groupby.groups(
+            [*self.obj._index._columns, *self.obj._columns]
+        )
+        grouped_keys = cudf.core.index._index_from_columns(grouped_key_cols)
+        grouped_values = self.obj._from_columns_like_self(
+            grouped_value_cols,
+            column_names=self.obj._column_names,
+            index_names=self.obj._index_names,
+        )
         group_names = grouped_keys.unique()
         return (group_names, offsets, grouped_keys, grouped_values)
 
-    def _normalize_aggs(self, aggs):
+    def _normalize_aggs(
+        self, aggs: Any
+    ) -> Tuple[Tuple[Any], Tuple[ColumnBase], List[List[Any]]]:
```

Comment:
Can the second `List[List[Any]]` be something like `List[List[Aggregation]]` instead? Not sure what the type name should be here, but we might be able to do better than `Any`.

Side note: For improved typing semantics, I'd love to define a type `ColumnName = Any` and use that throughout the code base. It might even allow us to locate places where our types would be invalid if we changed to `ColumnName = str` in the future.

Reply:
We can actually be precise here: per definition below, the aggregation type is 
`Union[str, List[str], Dict[ColumnName, List[str]]]`.

As for `ColumnName`, can we narrow it down to `Hashable`? Since we require it to act as keys in the column accessor. cc. @shwina @vyasr

Reply:
I think that's probably accurate, but in general we need to do a lot of working on _typing.py and that shouldn't be a blocker for this PR. We also define `ScalarLike = Any` and `ColumnLike = Any` at the moment. As the refactoring proceeds I think we should be adding new types incrementally as we solidify APIs, then hopefully we'll be at a point where we can do a thorough review to at least define all the types that we need to have (even if they are more broadly defined than they really should be).